tests/page/page-history.spec.ts

```diff
@@ -110,6 +110,14 @@ it('page.reload should work', async ({ page, server }) => {
   expect(await page.evaluate(() => window['_foo'])).toBe(undefined);
 });
 
+it('page.reload should not push history entry', async ({ page, server }) => {
+  it.info().annotations.push({ type: 'issue', description: 'https://github.com/microsoft/playwright/issues/22640' });
+  await page.goto(server.EMPTY_PAGE);
+  console.log(await page.evaluate('history.length'));
```

Comment:
assert?

Reply:
IIRC the Chromium build actually counts the initial blank page as one entry so you can't directly hardcode the length in. Should instead assert that history lengths are equal before and after the reload.